app/lib/ui/views/active/form_field/transporter_info_form_field.dart

```diff
@@ -0,0 +1,216 @@
+import 'package:app/core/models/address.dart';
+import 'package:app/core/models/transporter_info.dart';
+import 'package:app/core/utilities/optional.dart';
+import 'package:app/ui/views/active/form_field/dynamic_string_form_field.dart';
+import 'package:app/ui/views/active/form_field/string_form_field.dart';
+import 'package:app/ui/widgets/utility/date_time_picker.dart';
+import 'package:flutter/material.dart';
+
+import 'address_form_field.dart';
+
+class TransporterInfoFormField extends StatefulWidget {
+  final Function(TransporterInfo info) onSaved;
+  final TransporterInfo initialInfo;
+  final String title = "Transporter's Information";
+
+  // Use the form key to save all the fields of this form
+  final formKey = GlobalKey<FormState>();
+
+  TransporterInfoFormField(
+      {Key key, @required this.initialInfo, @required this.onSaved})
+      : super(key: key);
+
+  @override
+  _TransporterInfoFormFieldState createState() =>
+      _TransporterInfoFormFieldState();
+}
+
+class _TransporterInfoFormFieldState extends State<TransporterInfoFormField> {
+  String _companyName;
```

Comment:
Ugly as this is, this is the same as the ViewModel's I'm familiar with in `TornadoFx`. Gotta have that temporary object bits for caching.